src/coreclr/jit/morph.cpp

```diff
@@ -5562,48 +5572,61 @@ GenTree* Compiler::fgMorphArrayIndex(GenTree* tree)
 
     if (fgIsCommaThrow(tree))
     {
-        if ((arrElem != indTree) ||         // A new tree node may have been created
-            (indTree->OperGet() != GT_IND)) // The GT_IND may have been changed to a GT_CNS_INT
+        if ((arrElem != indTree) ||    // A new tree node may have been created
+            (indTree->OperIs(GT_IND))) // The GT_IND may have been changed to a GT_CNS_INT
```

Comment:
Shouldn't this be `!indTree->OperIs(GT_IND)` (i.e., you dropped the `!`)?

Reply:
oh, it explains some regressions 🤦‍♂️ thanks for noticing it!